test/index.js

```diff
@@ -33,25 +33,52 @@ function Just(x) {
   return {constructor: MaybeTypeRep, isNothing: false, isJust: true, value: x};
 }
 
+function mock($$type) {
+  return {constructor: {'@@type': $$type}};
+}
+
+function TypeIdentifier(namespace, name, version) {
+  return {namespace: namespace, name: name, version: version};
+}
+
+function named(name) {
+  return TypeIdentifier(null, name, 0);
+}
 
 test('type', function() {
-  eq(type(null), 'Null');
-  eq(type(undefined), 'Undefined');
-  eq(type({constructor: null}), 'Object');
-  eq(type({constructor: {'@@type': null}}), 'Object');
-  eq(type({constructor: {'@@type': new String('')}}), 'Object');
-  eq(type(Identity(42)), 'my-package/Identity');
-  eq(type(Identity), 'Function');
-  eq(type(Identity.prototype), 'Object');
-  eq(type(Nothing), 'my-package/Maybe');
-  eq(type(Just(0)), 'my-package/Maybe');
-  eq(type(Nothing.constructor), 'Object');
-
-  eq(type(false), 'Boolean');
-  eq(type(0), 'Number');
-  eq(type(''), 'String');
-
-  eq(type(new Boolean(false)), 'Boolean');
-  eq(type(new Number(0)), 'Number');
-  eq(type(new String('')), 'String');
+  eq(type(null), named('Null'));
+  eq(type(undefined), named('Undefined'));
+
+  eq(type({constructor: null}), named('Object'));
```

Comment:
We could use `Object.create(null)` here, to avoid GitHub mucking up the code-highlighting.